portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,22 +48,35 @@
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
         Query query = new Query(COMMENT_KIND).addSort(TIMESTAMP_PROPERTY, SortDirection.DESCENDING);
 
-        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        // Construct a read policy for eventual consistency
+        ReadPolicy policy = new ReadPolicy(ReadPolicy.Consistency.STRONG);
+
+        // Set the read policy
+        DatastoreServiceConfig strongConsistentConfig =
+            DatastoreServiceConfig.Builder.withReadPolicy(policy);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(strongConsistentConfig);
         PreparedQuery results = datastore.prepare(query);
 
         // Create ArrayList of comments and populate from the database
-        ArrayList<String> commentsList = new ArrayList<String>();
+        ArrayList<Comment> commentsList = new ArrayList<Comment>();
         Iterator resultIterator = results.asIterable().iterator();
         for (int i = 0; i < Integer.parseInt(request.getParameter(NUM_COMMENTS_PARAMETER)); i++) {
```

Comment:
Could you add some error checking here? In particular, NUM_COMMENTS_PARAMETER should be specified and must be a valid Integer. If not, you should return an error.

Reply:
wasn't sure if I could throw an exception so I just added a print statement for bad input